Progimg2

abanuog opened the pull request:
The program page template was changed to show a single hero image that has been added to the page. The Drupal backend 
was updated to include a new field for the Image in the Program content type.

Use the https://progimg-chug.pantheonsite.io URL as the Drupal source. 

From the list of Program pages on the front page:

- click Masters of Data Science to view a page with a hero image
- click Bachelor of Arts (or any other program) to view a page without a hero image
- click Studio Art MFA to view a page that uses the tagged image 

The "old" way of showing images has been commented out, so we can easily revert back to it.

snschau commented:
When completing this PR - will each Program Page need to be edited and have the image selected - as we are removing the tag method of selecting the hero image?

wsadmin commented:
> When completing this PR - will each Program Page need to be edited and have the image selected - as we are removing the tag method of selecting the hero image?

This makes me wonder if it's possible to use both methods - check for tagged image, and if null, check for feature image

abanuog commented:
> When completing this PR - will each Program Page need to be edited and have the image selected - as we are removing the tag method of selecting the hero image?

I guess we can ask the users to do that ?

tqureshi-uog commented:
I'd also like to understand why the Hero component wasn't used to render the feature image as it is in the Basic Page template.

abanuog commented:
> I'd also like to understand why the Hero component wasn't used to render the feature image as it is in the Basic Page template.

It's using a different query structure to get the image, and  the Hero component needs a specific one.

snschau commented:
> > When completing this PR - will each Program Page need to be edited and have the image selected - as we are removing the tag method of selecting the hero image?
> 
> I guess we can ask the users to do that ?

If it is something we change - we can ask them going forward to use the new method - but, we should make sure items work as they did before a change.

tqureshi-uog commented:
The more I look at this, the more I feel it can be handled without even touching the current image-header functions in program-page.js - examine how basic-page.js retrieves the feature image. The same approach might be possible in program-page.js by altering the GraphQL allMediaImage query with an additional filter as per https://dgraph.io/docs/graphql/queries/and-or-not/

In other words, allMediaImage on program-page.js, which currently only checks for a tag match, could be updated to look for either a tag match **or** an elemMatch as basic-page.js does.

abanuog commented:
- Added logic to show tagged images, if no page image was selected (can be tested using the Studio Art MFA page)
- Removed imageData from call to renderProgramStats as it wasn't being used
- Added the links to the fonts that James Harris wanted.
